### PR TITLE
Reintroduces the `test_end_to_end_repeated_transfers`. 

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

This test had some problems of duplicated notifications.

A possible origin of this class of error is the function `compute` which is a CAS function and so can be used several times if a notification A is sent at the same as notification B, the function can be called several times.

This is why the manual [papaya](https://docs.rs/papaya/latest/papaya/struct.HashMap.html#method.compute) recommends using a function that is pure. The function that we used and was emitting notifications is definitely not pure, nor idempotent.

On the other hand, duplicate notifications, do not have to be fatal to the running of an application.

## Proposal

We address the purity of the function by changing the code and emitting the notification out of the `compute` code. We also make the CI test resistant to duplicate notifications.

## Test Plan

CI.
The test `test_end_to_end_repeated_transfer` is reintroduced.

## Release Plan

The change could be backported to `testnet_conway`.

## Links

None

### CODE REVIEW

File: linera-core/src/notifier.rs [hunk 1]
```diff
@@ -63,22 +63,45 @@ where
...
+        // inside it.
+        let Some(senders) = pinned.get(chain_id).cloned() else {
+            trace!("Chain {chain_id} has no subscribers.");
+            return;
+        };
+
+        // Send notifications (side effect — must happen exactly once).
+        let mut has_dead = false;
+        for sender in &senders {
+            if sender.send(notification.clone()).is_err() {
```
Comment (afck) [inline linera-core/src/notifier.rs:80]: Actually, this _should_ fix the duplicate notifications, shouldn't it? Now that we're not calling `send` inside `compute` anymore, we don't expect any duplicates anymore? I'd not modify the test then, and see if it's still flaky.

Comment (MathieuDutSik) [reply linera-core/src/notifier.rs:80]: I agree that it should fix the duplicate notification.
I could of course revert the change in the test, but not that some traces would still be left of duplicate notification.

Comment (afck) [reply linera-core/src/notifier.rs:80]: I think as long as we don't know of a place in the code that could definitely cause duplicates, we should make the test fail if there's a duplicate.

Comment (MathieuDutSik) [reply linera-core/src/notifier.rs:80]: Fair enough. Let us try.

**Comment by afck** [comment]
@MathieuDutSik: Feel free to close https://github.com/linera-io/linera-protocol/issues/5722 once this is merged.

**Comment by MathieuDutSik** [comment]
> @MathieuDutSik: Feel free to close #5722 once this is merged.

Ok, done. It passed the CI.
Let us hope that it does not show up later in the future or in a different form.
